R/render-plot.R

```diff
@@ -266,24 +321,171 @@ drawPlot <- function(name, session, func, width, height, pixelratio, res, ...) {
   )
 }
 
+base_set_params <- function(bg, fg) {
+  params <- list()
+  if (!is.null(bg)) {
+    params <- c(params, par(bg = bg))
+  }
+  if (!is.null(fg)) {
+    params <- c(params, par(
+      fg = fg,
+      col.axis = fg,
+      col.lab = fg,
+      col.main = fg,
+      col.sub = fg
+    ))
+  }
+  params
+}
+
+grid_set_params <- function(bg, fg) {
+  grid::gpar(fill = bg, col = fg)
+}
+
+lattice_set_params <- function(bg, fg) {
+  if (system.file(package = "lattice") == "") return()
+  old_par <- lattice::trellis.par.get()
+  lattice::trellis.par.set(
+    # See figure 9.3 for an example of where grid gpar matters
+    # http://lmdvr.r-forge.r-project.org/figures/figures.html
+    grid.pars = list(col = fg, fill = bg),
+    background = list(col = bg),
+    panel.background = list(col = setAlpha(fg, 0.2)),
+    add.line = list(col = fg),
+    add.text = list(col = fg),
+    plot.polygon = list(border = fg, col = bg),
+    box.dot = list(col = fg),
+    box.rectangle = list(col = fg, fill = bg),
+    box.umbrella = list(col = fg),
+    dot.line = list(col = setAlpha(fg, 0.2)),
+    dot.symbol = list(col = fg),
+    plot.line = list(col = fg),
+    plot.symbol = list(col = fg),
+    reference.line = list(col = bg),
+    strip.background = list(col = setAlpha(fg, 0.4)),
+    strip.border = list(col = "transparent"),
+    superpose.polygon = list(border = rep("transparent", 7)),
+    axis.line = list(col = fg),
+    axis.text = list(col = fg),
+    box.3d = list(col = fg),
+    par.xlab.text = list(col = fg),
+    par.ylab.text = list(col = fg),
+    par.zlab.text = list(col = fg),
+    par.main.text = list(col = fg),
+    par.sub.text = list(col = fg)
+  )
+  old_par
+}
+
+lattice_set_par_list <- function(params) {
+  if (system.file(package = "lattice") == "") return()
+  do.call(lattice::trellis.par.set, params)
+}
+
+maybe_set_palette <- function(fg) {
+  p <- palette()
+  if (is_default_pallete(p)) {
+    p[[1]] <- fg
+    palette(p)
+  }
+  p
```

Comment:
Is this supposed to return the old palette, or the modified one? A comment where this is called seems to indicate it's the former, but the implementation seems to be doing the latter.

Reply:
Oops, yea, it's supposed to return the old one, thanks for fixing.